app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/home/birthday/BirthdayViewModel.kt

```diff
@@ -33,68 +35,147 @@ class BirthdayViewModel(
+    fun setDate(dateUtc: String) {
+        _dateUtc.update { dateUtc }
+        _dateDayMonthYear.update { dateUtc.fromUtcToDayMonthYearDate() }
     }
 
     fun setCreateBirthdayErrorFalse() {
         _createBirthdayError.value = false
     }
 
-    fun createBirthday(navigateToHomeScreen: () -> Unit) {
+    fun setUpdateBirthdayErrorFalse() {
+        _updateBirthdayError.value = false
+    }
+
+    fun setDeleteBirthdayErrorFalse() {
+        _deleteBirthdayError.value = false
+    }
+
+    fun setBirthday(birthday: Birthday?) {
+        _editModeBirthday.update { birthday }
+        if (birthday != null) {
+            setName(birthday.name)
+            setImage(birthday.image)
+            setRelationship(birthday.relation)
+            setDate(birthday.dateUtc)
+        } else {
+            setName("")
+            setImage(null)
+            setRelationship(null)
+            _dateUtc.update { Calendar.getInstance().timeInMillis.fromMillisToUtcDate() }
+            _dateDayMonthYear.update { "__.__.____" }
+        }
+    }
+
+    fun createBirthday(
+        navigateToHomeScreen: () -> Unit,
+        onCompletion: () -> Unit,
+    ) {
         viewModelScope.launch(Dispatchers.IO) {
-            val imageUri = imageUri.value
+            val imageByteArray = imageByteArray.value
             birthdayRepository.createBirthday(
                 createBirthday = CreateBirthdayEntity(
                     name = _name.value,
-                    imageBase64 = imageUri?.uriToBase64(context = getApplication<Application>().applicationContext),
+                    imageBase64 = imageByteArray?.uriToBase64(),
                     relation = getApplication<Application>().applicationContext.getString(
                         _relationship.value?.resId ?: RelationshipEnum.BEST_FRIEND.resId
                     ),
-                    date = _date.value.toString(),
+                    dateUtc = _dateUtc.value,
                 ),
             ).onEach {
                 it.onSuccess { navigateToHomeScreen() }
                 it.onFailure { _createBirthdayError.update { true } }
-            }.catch {
-                _createBirthdayError.update { true }
-            }.collect()
+            }.onCompletion { onCompletion() }
+                .catch { _createBirthdayError.update { true } }
+                .flowOn(Dispatchers.Main).collect()
+        }
+    }
+
+    fun updateBirthday(
+        id: Int,
+        navigateToHomeScreen: () -> Unit,
+        onCompletion: () -> Unit,
+    ) {
+        viewModelScope.launch(Dispatchers.IO) {
+            val imageByteArray = imageByteArray.value
+            birthdayRepository.updateBirthday(
+                id = id, updateBirthdayEntity = UpdateBirthdayEntity(
+                    name = _name.value,
+                    imageBase64 = imageByteArray?.uriToBase64(),
+                    relation = getApplication<Application>().applicationContext.getString(
```

Comment:
"getApplication<Application>().applicationContext" could  be defined in scope of view model